core/js/typespecs/functions.ts

```diff
@@ -1,50 +1,138 @@
+  .union([
+    z
+      .object({
+        function_id: z.string().describe("The ID of the function"),
+        version: z.string().optional().describe("The version of the function"),
+      })
+      .describe("Function id"),
+    z
+      .object({
+        project_name: z
+          .string()
+          .describe("The name of the project containing the function"),
+        slug: z.string().describe("The slug of the function"),
+        version: z.string().optional().describe("The version of the function"),
+      })
+      .describe("Project name and slug"),
+    z
+      .object({
+        global_function: z
+          .string()
+          .describe(
+            "The name of the global function. Currently, the global namespace includes the functions in autoevals",
+          ),
+      })
+      .describe("Global function name"),
+    z
+      .object({
+        prompt_session_id: z.string().describe("The ID of the prompt session"),
+        prompt_session_function_id: z
+          .string()
+          .describe("The ID of the function in the prompt session"),
+        version: z.string().optional().describe("The version of the function"),
+      })
+      .describe("Prompt session id"),
+  ])
+  .describe("Options for identifying a function");
+export type FunctionId = z.infer<typeof functionIdSchema>;
 
-export const functionIdSchema = z.union([
-  z.object({
-    function_id: z.string().describe("The ID of the function."),
-    version: z.string().optional().describe("The version of the function."),
-  }),
-  z.object({
-    project_name: z
+export const useFunctionSchema = functionIdSchema;
+
+export const invokeFunctionNonIdArgsSchema = z.object({
+  input: z
+    .any()
+    .optional()
+    .describe(
+      "Argument to the function, which can be any JSON serializable value",
+    ),
+  parent: z
+    .union([
+      z
+        .object({
+          object_type: z.enum(["project_logs", "experiment"]),
+          object_id: z
+            .string()
+            .describe("The id of the container object you are logging to"),
+        })
+        .describe("Object type and id"),
+      z
+        .string()
+        .optional()
+        .describe(
+          "The parent's span identifier, created by calling `.export()` on a span",
+        ),
+    ])
+    .describe("Options for tracing the function call"),
+  stream: z
+    .boolean()
+    .optional()
+    .describe(
+      "Whether to stream the response. If true, results will be returned in the Braintrust SSE format.",
+    ),
+});
+
+export const invokeFunctionSchema = useFunctionSchema.and(
+  invokeFunctionNonIdArgsSchema,
+);
+export type InvokeFunctionRequest = z.infer<typeof invokeFunctionSchema>;
+
+export const invokeApiSchema = invokeFunctionNonIdArgsSchema
+  .merge(
+    z.object({
+      version: z.string().optional().describe("The version of the function"),
+    }),
+  )
+  .describe("The request to invoke a function");
+
+export const runEvalSchema = z
+  .strictObject({
```

Comment:
replace with `.object` and remove `.strip`?

Reply:
yep